project_stock/models/project_project.py

```diff
@@ -32,6 +32,13 @@ class ProjectProject(models.Model):
     )
 
     @api.onchange("picking_type_id")
-    def onchange_picking_type(self):
+    def _onchange_picking_type_id(self):
         self.location_id = self.picking_type_id.default_location_src_id.id
         self.location_dest_id = self.picking_type_id.default_location_dest_id.id
+
+    def write(self, vals):
+        res = super().write(vals)
```

Comment:
Remember to always put proper docstrings:
```suggestion
        """Update location information on pending moves when changed."""
        res = super().write(vals)
```